server/issue.go

```diff
@@ -20,6 +20,56 @@ import (
 	"github.com/mattermost/mattermost-plugin-jira/server/utils"
 )
 
+func makePost(userId, channelId, message string) *model.Post {
+	return &model.Post{
+		UserId:    userId,
+		ChannelId: channelId,
+		Message:   message,
+	}
+}
+
+func httpAPITransitionIssue(ji Instance, w http.ResponseWriter, r *http.Request) (int, error) {
+	requestData := model.PostActionIntegrationRequestFromJson(r.Body)
+
+	mattermostUserId := requestData.UserId
+	if mattermostUserId == "" {
+		return http.StatusUnauthorized, errors.New("not authorized")
+	}
+
+	issueKey := requestData.Context["issueKey"].(string)
+	toState := requestData.Context["selected_option"].(string)
+
+	plugin := ji.GetPlugin()
+	jiraBotID := plugin.getUserID()
+	channelID := requestData.ChannelId
+
+	msg, err := plugin.transitionJiraIssue(mattermostUserId, issueKey, toState)
+	if err != nil {
+		_ = plugin.API.SendEphemeralPost(mattermostUserId, makePost(jiraBotID, channelID, "Failed to transition this issue."))
+		return http.StatusInternalServerError, err
+	}
+
+	jiraUser, err := plugin.userStore.LoadJIRAUser(ji, mattermostUserId)
+	if err != nil {
+		_ = plugin.API.SendEphemeralPost(mattermostUserId, makePost(jiraBotID, channelID, "Your username is not connected to Jira. Please type `jira connect`."))
```

Comment:
Doesn't make much sense to me that this would fail after the transition succeeds. Maybe put this block above the one above?